### PR TITLE
✨ Source Github: add `files` and `stats` propertied to `pull_request_commits` schema

### LINKED ISSUE #34318: [source-github] Add `files` property to `commits` stream schema
### Topic

Add new property to Source Github

### Relevant information

As documented in [Github's commits REST API](https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#list-commits), one of the properties that is available when listing commits is `files`, which contains relevant information about each file changed within a commit (number of additions, deletions, etc).

I believe this property could be added to the Source Github, so I created this [PR](https://github.com/airbytehq/airbyte/pull/34317) to do so.

As it's my first contribution to Airbyte, I may be missing something, but I tried to strictly follow the contribution documents.

### PR DESCRIPTION
**Pull request by dlouseiro**
## What

The current implementation of the `commits` stream in the source Github is missing a property called `files`, documented in the response schema of Github's REST API for commits ([here](https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#list-commits)).

This property is an array, containing detailed information on the changes done in each file within the commit (number of additions, number of removals, if the file was added or removed, etc).

This PR resolves #34318 

## How

Added a simplified version of the response schema provided by Github in `airbyte-integrations/connectors/source-github/source_github/schemas/commits.json`.

## 🚨 User Impact 🚨

As this PR simply adds a property to an existing stream, it should correspond to a minor version of the connector.

### CODE REVIEW

**Comment by dlouseiro** [comment]
Resolved #34317

File: airbyte-integrations/connectors/source-github/source_github/schemas/commits.json [hunk 1]
```diff
@@ -121,6 +121,47 @@
           }
         }
       }
+    },
+    "files": {
+      "type": ["null", "array"],
+      "items": {
+        "type": ["null","object"],
+        "properties": {
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-github/source_github/schemas/commits.json:129]: @dlouseiro it's better to add the `additionalProperties: true` so in cases there is a missing or new property the connector won't break.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Small detail, can you bump the metadata.yaml and changelog version in the doc? Take a look in the guidelines here: https://docs.airbyte.com/contributing-to-airbyte/change-cdk-connector#code-your-contribution

File: airbyte-integrations/connectors/source-github/source_github/schemas/commits.json (hunk 1 above)
Comment (dlouseiro) [reply airbyte-integrations/connectors/source-github/source_github/schemas/commits.json:129]: Sure, makes sense! Will do

**Comment by dlouseiro** [comment]
> Small detail, can you bump the metadata.yaml and changelog version in the doc? Take a look in the guidelines here: https://docs.airbyte.com/contributing-to-airbyte/change-cdk-connector#code-your-contribution

Sure, will do!

**Comment by dlouseiro** [comment]
@marcosmarxm , in the meantime I did some extra tests and noticed that in order to fetch the `files` property for the `commits` stream we'd need to do it [commit by commit](https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#get-a-commit) as this property is not available in the API to [list commits](https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#list-commits).

Although, the same `files` as well as `stats` properties are available when fetching [Pull request commits](https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#list-commits-on-a-pull-request) so I removed them from the `commits` stream and added them to the `pull_request_commits` stream.

**Comment by lazebnyi** [comment]
@dlouseiro Thanks for your PR. But pull requests commits schema has not field files - https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#list-commits-on-a-pull-request

**Comment by dlouseiro** [comment]
> @dlouseiro Thanks for your PR. But pull requests commits schema has not field files - https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#list-commits-on-a-pull-request 

hey @lazebnyi youre completely right. I meant to close this pr in the meantime. The files and stats properties are listed in the response schema of the list pr commits API, but are actually not sent in the payload.

I based the or on the response schema documented by GitHub but when tried to test it did not receive the files not stats properties. Thanks for closing it

**Pull request closed without merging**